# 01 : Acquisition des données

Ce notebook couvre la première étape du projet : le chargement et la découverte 
du dataset principal.

**Source :** Kaggle : Mental Health and Lifestyle Habits 2019-2024  
**Fichier :** `Mental_Health_Lifestyle_Dataset.csv`  
**Objectif :** Charger les données, comprendre leur structure, identifier 
les premières pistes de nettoyage.

In [3]:
# Manipulation des données
import pandas as pd

# Gestion des chemins de fichiers (portable Windows/Mac/Linux)
import os

# On remonte à la racine du projet depuis le dossier notebooks/
# Ainsi tous les chemins relatifs fonctionneront depuis la racine
os.chdir("..")

# Vérification — doit afficher le chemin racine du projet
print(f"Répertoire de travail : {os.getcwd()}")

Répertoire de travail : c:\Users\PC\depression-lifestyle-predictor


In [4]:
# Chemin relatif vers le fichier CSV (fonctionne depuis la racine du projet)
data_path = os.path.join("data", "raw", "Mental_Health_Lifestyle_Dataset.csv")

# Chargement du CSV dans un DataFrame pandas

df = pd.read_csv(data_path, sep=",")

print(f"Dataset chargé avec succès : {df.shape[0]} lignes, {df.shape[1]} colonnes")

Dataset chargé avec succès : 3000 lignes, 12 colonnes


In [5]:
# 1. Afficher les 5 premières lignes pour visualiser la structure
print("=== APERÇU DES PREMIÈRES LIGNES ===")
display(df.head())

# 2. Dimensions du dataset
print(f"\n=== DIMENSIONS ===")
print(f"Nombre de lignes    : {df.shape[0]}")
print(f"Nombre de colonnes  : {df.shape[1]}")

#  3. Types de données et valeurs non-nulles par colonne
print("\n=== TYPES DE DONNÉES ===")
print(df.dtypes)

=== APERÇU DES PREMIÈRES LIGNES ===


,Country,Age,Gender,Exercise Level,Diet Type,Sleep Hours,Stress Level,Mental Health Condition,Work Hours per Week,Screen Time per Day (Hours),Social Interaction Score,Happiness Score
0,Brazil,48,Male,Low,Vegetarian,6.3,Low,NaN,21,4.0,7.8,6.5
1,Australia,31,Male,Moderate,Vegan,4.9,Low,PTSD,48,5.2,8.2,6.8
2,Japan,37,Female,Low,Vegetarian,7.2,High,NaN,43,4.7,9.6,9.7
3,Brazil,35,Male,Low,Vegan,7.2,Low,Depression,43,2.2,8.2,6.6
4,Germany,46,Male,Low,Balanced,7.3,Low,Anxiety,35,3.6,4.7,4.4



=== DIMENSIONS ===
Nombre de lignes    : 3000
Nombre de colonnes  : 12

=== TYPES DE DONNÉES ===
Country                            str
Age                              int64
Gender                             str
Exercise Level                     str
Diet Type                          str
Sleep Hours                    float64
Stress Level                       str
Mental Health Condition            str
Work Hours per Week              int64
Screen Time per Day (Hours)    float64
Social Interaction Score       float64
Happiness Score                float64
dtype: object


In [6]:
# Calcul du nombre de valeurs manquantes par colonne
missing_count = df.isnull().sum()

# Calcul du pourcentage de valeurs manquantes par colonne
missing_percent = (df.isnull().sum() / len(df)) * 100

# On regroupe dans un DataFrame lisible, trié par ordre décroissant
missing_df = pd.DataFrame({
    "Valeurs manquantes" : missing_count,
    "Pourcentage (%)"    : missing_percent.round(2)
}).sort_values("Valeurs manquantes", ascending=False)

print("=== VALEURS MANQUANTES PAR COLONNE ===")
display(missing_df)

=== VALEURS MANQUANTES PAR COLONNE ===


,Valeurs manquantes,Pourcentage (%)
Mental Health Condition,595,19.83
Country,0,0.00
Gender,0,0.00
Age,0,0.00
Exercise Level,0,0.00
Diet Type,0,0.00
Sleep Hours,0,0.00
Stress Level,0,0.00
Work Hours per Week,0,0.00
Screen Time per Day (Hours),0,0.00


In [7]:
# Statistiques descriptives pour les colonnes numériques
print("=== STATISTIQUES DESCRIPTIVES ===")
display(df.describe().round(2))

=== STATISTIQUES DESCRIPTIVES ===


,Age,Sleep Hours,Work Hours per Week,Screen Time per Day (Hours),Social Interaction Score,Happiness Score
count,3000.00,3000.00,3000.00,3000.00,3000.00,3000.00
mean,41.23,6.48,39.47,5.09,5.47,5.40
std,13.43,1.50,11.45,1.75,2.56,2.56
min,18.00,1.40,20.00,2.00,1.00,1.00
25%,30.00,5.50,30.00,3.60,3.30,3.20
50%,41.00,6.50,39.00,5.10,5.50,5.40
75%,53.00,7.50,50.00,6.60,7.60,7.50
max,64.00,11.30,59.00,8.00,10.00,10.00


In [8]:
# Sélection des colonnes de type "object" (texte/catégories)
categorical_cols = df.select_dtypes(include="object").columns

print("=== VALEURS UNIQUES PAR COLONNE CATÉGORIELLE ===\n")

for col in categorical_cols:
    unique_vals = df[col].unique()
    print(f" {col} ({len(unique_vals)} valeurs uniques) :")
    print(f"   {unique_vals}\n")

=== VALEURS UNIQUES PAR COLONNE CATÉGORIELLE ===

 Country (7 valeurs uniques) :
   <ArrowStringArray>
['Brazil', 'Australia', 'Japan', 'Germany', 'India', 'Canada', 'USA']
Length: 7, dtype: str

 Gender (3 valeurs uniques) :
   <ArrowStringArray>
['Male', 'Female', 'Other']
Length: 3, dtype: str

 Exercise Level (3 valeurs uniques) :
   <ArrowStringArray>
['Low', 'Moderate', 'High']
Length: 3, dtype: str

 Diet Type (5 valeurs uniques) :
   <ArrowStringArray>
['Vegetarian', 'Vegan', 'Balanced', 'Junk Food', 'Keto']
Length: 5, dtype: str

 Stress Level (3 valeurs uniques) :
   <ArrowStringArray>
['Low', 'High', 'Moderate']
Length: 3, dtype: str

 Mental Health Condition (5 valeurs uniques) :
   <ArrowStringArray>
[nan, 'PTSD', 'Depression', 'Anxiety', 'Bipolar']
Length: 5, dtype: str



C:\Users\PC\AppData\Local\Temp\ipykernel_18316\1798684880.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_cols = df.select_dtypes(include="object").columns


## Conclusions de l'acquisition

-Le dataset contient **3000 lignes** et **12 colonnes**
- 6 colonnes catégorielles : Country, Gender, Exercise Level, 
  Diet Type, Stress Level, Mental Health Condition
- 6 colonnes numériques : Age, Sleep Hours, Work Hours per Week, 
  Screen Time per Day, Social Interaction Score, Happiness Score
- **Seule valeur manquante :** Mental Health Condition avec 595 NaN 
(19.83%) c'est notre variable cible, à traiter en priorité 
  dans le nettoyage
- Prochaine étape : nettoyage des données (`02_data_cleaning.ipynb`)